pkg/list/values/extract.go

```diff
@@ -0,0 +1,308 @@
+package values
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/cloudposse/atmos/pkg/list/errors"
+	"github.com/cloudposse/atmos/pkg/utils"
+)
+
+// handleSpecialComponent processes special components like settings and metadata.
+func handleSpecialComponent(stack map[string]interface{}, component string) (map[string]interface{}, bool) {
+	// First check if the component exists at the top level
+	if section, ok := stack[component].(map[string]interface{}); ok {
+		return section, true
+	}
+
+	// If not found at the top level and component is "settings", look for it in components
+	if component == "settings" {
```

Comment:
@aknysh can we have `component.terraform.settings`?

Reply:
This is wrong. 

It would be a top-level key of `terraform.settings`, not `component.terraform.settings`